src/mongocxx/test/search_index_view.cpp

```diff
@@ -1 +1,211 @@
+#include <chrono>
+#include <thread>
+
+#include <bsoncxx/builder/basic/document.hpp>
+#include <bsoncxx/builder/basic/kvp.hpp>
+#include <bsoncxx/oid.hpp>
+#include <bsoncxx/test_util/catch.hh>
+#include <mongocxx/client.hpp>
+#include <mongocxx/cursor.hpp>
+#include <mongocxx/instance.hpp>
 #include <mongocxx/search_index_view.hpp>
+#include <mongocxx/test_util/client_helpers.hh>
+
+namespace {
+using bsoncxx::builder::basic::kvp;
+using bsoncxx::builder::basic::make_document;
+using namespace mongocxx;
+
+bool does_search_index_exist_on_cursor(cursor& c, search_index_model& model, bool with_status) {
+    for (auto&& doc : c) {
+        // check the name, that the index is queryable, and that the definition matches.
+        if (doc["name"].get_string().value == *model.name() && doc["queryable"].get_bool().value &&
+            doc["latestDefinition"].get_document().view() == model.definition()) {
+            // optional addition check needed if with_status is set
+            if (!with_status || (with_status && bsoncxx::string::to_string(
+                                                    doc["status"].get_string().value) == "READY")) {
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+bool wait_for_search_index(search_index_view& siv,
+                           search_index_model& model,
+                           bool to_exist,
+                           bool with_status) {
+    auto c = siv.list();
+
+    // 5 minutes before timeout
+    for (int i = 0; i < 60; ++i) {
+        if (does_search_index_exist_on_cursor(c, model, with_status) == to_exist)
+            return true;
+        std::this_thread::sleep_for(std::chrono::seconds(5));
+        c = siv.list();
+    }
+    return false;
+}
+
+TEST_CASE("atlas search indexes prose tests", "") {
+    instance::current();
+
+    auto uri_getenv = std::getenv("MONGODB_URI");
+    if (!uri_getenv) {
+        WARN("Skipping - Test requires the environment variable: MONGODB_URI");
+        return;
+    }
+
+    client mongodb_client{uri{uri_getenv}};
+
+    database db = mongodb_client[std::getenv("FUNCTION_NAME")];
+
+    SECTION("create one with name and definition") {
+        // use a randomly generated collection name as there's a server side limitation that
+        // prevents multiple search indexes with the same name, definition and collection name
+        // from being created.
+        bsoncxx::oid id;
+        db.create_collection(id.to_string());
+        collection coll = db[id.to_string()];
```

Comment:
```suggestion
        auto coll = db.create_collection(id.to_string());
```

Suggested simplification. Also applies to other tests.